packages/flutter/lib/src/painting/image_resolution.dart

```diff
@@ -398,3 +407,120 @@ class AssetImage extends AssetBundleImageProvider {
   @override
   String toString() => '${objectRuntimeType(this, 'AssetImage')}(bundle: $bundle, name: "$keyName")';
 }
+
+
+/// Fetches an image from an [AssetBundle], associating it with the given scale.
+///
+/// This implementation requires an explicit final [assetName] and [scale] on
+/// construction, and ignores the device pixel ratio and size in the
+/// configuration passed into [resolve]. For a resolution-aware variant that
+/// uses the configuration to pick an appropriate image based on the device
+/// pixel ratio and size, see [AssetImage].
+///
```

Comment:
Can we document here what happens if no asset variant with that exact scale specified exists?